cypress/support/POM/Client_PO.js

```diff
@@ -0,0 +1,107 @@
+class Client {
+  typeFirstName(index, firstname) {
+    cy.get("input[name='first_name']").eq(index).clear().type(firstname);
+
+    return this;
+  }
+
+  editFirstName(index, Firstname) {
+    cy.get("#first_name").eq(index).clear().type(Firstname);
+
+    return this;
+  }
+
+  editLastName(index, Lastname) {
+    cy.get("#last_name").eq(index).clear().type(Lastname);
+
+    return this;
+  }
+
+  typeLastName(index, Lastname) {
+    cy.get("input[name='last_name']").eq(index).clear().type(Lastname);
+
+    return this;
+  }
+
+  clickOnClientButton() {
+    cy.get("#clientsMenu").click();
+
+    return this;
+  }
+
+  clickOnAddButton() {
+    cy.get(".blueButton.button.field").click();
+
+    return this;
+  }
+
+  clickOnSaveButton() {
+    cy.get(".pull-right .blueButton.button").contains("Save").click();
+
+    return this;
+  }
+
+  selectAssignee() {
+    cy.get("div[name='assignee'] > div[role='combobox'] i").dblclick();
+    cy.get(".form li").first().click();
+
+    return this;
+  }
+
+  selectProduct() {
+    cy.get("div[name='selectProducts'] > div[role='combobox']").click({force:true});
+    cy.get("ul[role='listbox'] > li").eq(2).click({force:true});
+    cy.get("div[name='selectProducts'] > div[role='combobox'] i").click({force:true});
+
+    return this;
+  }
+
+  clickOnCancelButton() {
+    cy.get(".pull-right .defaultButton.button").contains("Cancel").click();
+
+    return this;
+  }
+
+  typeEmail(index, email) {
+    cy.get("input[name='email']").eq(index).clear().type(email);
+
+    return this;
+  }
+
+  editEmail(index, email) {
+    cy.get("#email").eq(index).clear().type(email);
```

Comment:
eq(index) might not be needed here .You could edit without taking index in all edit form fields